nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs

```diff
@@ -160,39 +158,61 @@ async Task TrackOriginalContentsAsync(string directory, string fileName)
         }
 
         // do update
-        var updateOperations = GetUpdateOperations(discoveryResult).ToArray();
-        var allowedUpdateOperations = updateOperations.Where(u => IsUpdateAllowed(job, u.Dependency)).ToArray();
-
-        // requested update isn't listed => SecurityUpdateNotNeeded
-        var expectedSecurityUpdateDependencyNames = job.SecurityAdvisories
+        var existingPullRequests = job.GetAllExistingPullRequests();
+        var unhandledPullRequestDependenciesSet = existingPullRequests
+            .Select(pr => pr.Item2.Select(d => d.DependencyName).ToHashSet(StringComparer.OrdinalIgnoreCase))
+            .ToHashSet();
+        var remainingSecurityIssues = job.SecurityAdvisories
             .Select(s => s.DependencyName)
             .ToHashSet(StringComparer.OrdinalIgnoreCase);
-        var actualUpdateDependencyNames = allowedUpdateOperations
-            .Select(u => u.Dependency.Name)
-            .ToHashSet(StringComparer.OrdinalIgnoreCase);
-        var expectedDependencyUpdateMissingInActual = expectedSecurityUpdateDependencyNames
-            .Except(actualUpdateDependencyNames, StringComparer.OrdinalIgnoreCase)
-            .OrderBy(d => d, StringComparer.OrdinalIgnoreCase)
-            .ToArray();
-
-        foreach (var missingSecurityUpdate in expectedDependencyUpdateMissingInActual)
-        {
-            await _apiHandler.RecordUpdateJobError(new SecurityUpdateNotNeeded(missingSecurityUpdate));
-        }
+        var updateOperations = GetUpdateOperations(discoveryResult).ToArray();
 
-        foreach (var updateOperation in allowedUpdateOperations)
+        foreach (var updateOperation in updateOperations)
```

Comment:
This main loop is a bit more complicated now because we need to report certain types of PR messages immediately and other types later once we've done some update analysis.  In either case we need to track that we've already reported on that PR so we don't report on it again at the end, where we need to report on anything not reported on yet.